File: verifier-core/src/main/java/com/jetbrains/pluginverifier/verifiers/clazz/ClassVerifier.java
package com.jetbrains.pluginverifier.verifiers.clazz;

import com.intellij.structure.resolvers.Resolver;
import com.jetbrains.pluginverifier.api.VContext;
import org.objectweb.asm.tree.ClassNode;

public interface ClassVerifier {
  void verify(ClassNode clazz, Resolver resolver, VContext ctx);
}

File: verifier-core/src/main/java/com/jetbrains/pluginverifier/verifiers/clazz/ClassVerifier.kt
package com.jetbrains.pluginverifier.verifiers.clazz

import com.jetbrains.pluginverifier.verifiers.VerificationContext
import com.jetbrains.pluginverifier.verifiers.resolution.ClassFile

interface ClassVerifier {
  fun verify(classFile: ClassFile, context: VerificationContext)
}
